applications/vanilla/controllers/class.postcontroller.php

```diff
@@ -172,17 +172,22 @@ public function discussion($CategoryUrlCode = '') {
 
         touchValue('Type', $this->Data, 'Discussion');
 
-        // See if we should hide the category dropdown.
-        if ($this->ShowCategorySelector) {
+        if (!$UseCategories || $this->ShowCategorySelector) {
+            // See if we should fill the CategoryID value.
             $AllowedCategories = CategoryModel::getByPermission(
                 'Discussions.Add',
                 $this->Form->getValue('CategoryID', $this->CategoryID),
                 ['Archived' => 0, 'AllowDiscussions' => 1],
                 ['AllowedDiscussionTypes' => $this->Data['Type']]
             );
-            if (count($AllowedCategories) == 1) {
-                $AllowedCategory = array_pop($AllowedCategories);
+            $nbsAllowedCategories = count($AllowedCategories);
```

Comment:
`$AllowedCategoriesCount` is fine. Or allowedCount. Or categoryCount. Or the inverse of any of those.